libraries/ride-fuzzer-lib/src/main/java/com/adobe/ride/libraries/fuzzer/MetadataFuzzer.java

```diff
@@ -140,45 +148,142 @@ private void initializeFuzzer(String serviceName, ModelObject objectToBeFuzzed,
 
     // Populate global properties
     this.entity = objectToBeFuzzed;
+    this.entity.buildValidModelInstance();
+
     this.requestMethod = method;
     this.serviceName = serviceName;
-    this.properties = this.entity.getModelProperties();
-    this.instance = this.entity.getObjectMetadata();
+    this.properties = ModelObject.getObjectNodeProperties(entity, this.entity.getModel());
     this.requestMethod = method;
     this.contentType = contentType;
     this.requestBuilder = requestBuilder;
+    
+    if (requestBuilder != null) {
+      this.requestBuilder = requestBuilder;
+    } else {
+      this.requestBuilder = RestApiController.getRequestBuilder(false);
+      this.requestBuilder.addHeader("Content-Type", "application/json");
+    }
+
     this.filters = filters;
-    propertiesFuzzSet = new Object[properties.keySet().size()][3];
 
     // Prep TestNG dataprovider to make reporting better.
-    populateFuzzSet(properties, propertiesFuzzSet, "rootProperty");
+    root = (objectToBeFuzzed.getModelType() == ModelPropertyType.OBJECT) ? File.separator
+        : File.separator + "0" + File.separator ;
+
+    propertiesFuzzSet = populateFuzzSet(this.entity, new ArrayList<Object[]>(), root, this.entity.getModel());
```

Comment:
Need to change the signature here.  In order to recurs, need to work with ArrayList.